rubicon_ml/client/mixin.py

```diff
@@ -326,17 +333,61 @@ def dataframes(self, tags=[], qtype="or"):
             The dataframes previously logged to this client object.
         """
         project_name, experiment_id = self._get_parent_identifiers()
-        dataframes = [
-            client.Dataframe(d, self)
-            for d in self.repository.get_dataframes_metadata(
-                project_name, experiment_id=experiment_id
-            )
-        ]
-
+        if name is not None:
+            dataframes = [
+                client.Dataframe(d, self)
+                for d in self.repository.get_dataframes_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+                if d.name == name
+            ]
+        else:
+            dataframes = [
+                client.Dataframe(d, self)
+                for d in self.repository.get_dataframes_metadata(
+                    project_name, experiment_id=experiment_id
+                )
+            ]
         self._filter_dataframes(dataframes, tags, qtype)
 
         return self._dataframes
 
+    def dataframe(self, name=None, id=None):
+        """
+        Get the dataframe logged to this client object.
+
+        Parameters
+        ----------
+        id : str
+            The id of the artifact to get.
+        name : str
+            The name of the artifact to get.
+        Returns
+        -------
+        rubicon.client.Dataframe
+            The dataframe logged to this project with id `id` or name 'name'.
+        """
+        if (name is None and id is None) or (name is not None and id is not None):
+            raise ValueError("`name` OR `id` required.")
+
+        elif name is not None:
+            dataframes = self.dataframes(name=name)
+            if len(dataframes) == 0:
```

Comment:
I think changing the location of this Rubicon exception to the `dataframes` function would probably be better so it can then throw and exception for the circumstance where the name passed to both `dataframe` and `dataframes` don't exist. I did that in the `artifacts` function as well for reference.